Oooh interesting, these failures failures seem worth addressing.

```python
FAILED test/test_fork.py::TestFork::test_lock_client - DeprecationWarning: This process (pid=3147) is multi-threaded, use of fork() may lead to deadlocks in the child.
FAILED test/test_fork.py::TestFork::test_lock_object_id - DeprecationWarning: This process (pid=3147) is multi-threaded, use of fork() may lead to deadlocks in the child.
FAILED test/test_fork.py::TestFork::test_topology_reset - DeprecationWarning: This process (pid=3147) is multi-threaded, use of fork() may lead to deadlocks in the child.
```

I *think* we can safely ignore those warnings unless the same client is used across threads, assuming the monitor threads get re-created during the fork.

--- a/.github/workflows/test-python.yml
+++ b/.github/workflows/test-python.yml
@@ -59,7 +59,7 @@ jobs:
       matrix:
         # Tests currently only pass on ubuntu on GitHub Actions.
         os: [ubuntu-latest]
-        python-version: ["3.10", "pypy-3.11", "3.13t"]
+        python-version: ["3.10", "pypy-3.11", "3.13t", "3.15"]
         mongodb-version: ["8.0"]
 
     name: CPython ${{ matrix.python-version }}-${{ matrix.os }}

--- a/.github/workflows/test-python.yml
+++ b/.github/workflows/test-python.yml
@@ -62,7 +62,7 @@ jobs:
         python-version: ["3.10", "pypy-3.11", "3.13t", "3.15"]
         mongodb-version: ["8.0"]
 
-    name: CPython ${{ matrix.python-version }}-${{ matrix.os }}
+    name: Python ${{ matrix.python-version }}-${{ matrix.os }}
     steps:
       - uses: actions/checkout@v6.0.2
         with:

--- a/test/test_fork.py
+++ b/test/test_fork.py
@@ -36,6 +36,10 @@ from bson.objectid import ObjectId
     is_greenthread_patched(),
     "gevent does not support POSIX-style forking.",
 )
+@unittest.skipIf(
+    sys.version_info >= (3, 15),
+    "fork() in multi-threaded processes is deprecated in Python 3.15+ (PYTHON-5874)",
+)
 class TestFork(IntegrationTest):
     def test_lock_client(self):
         # Forks the client with some items locked.

